core/services/registrysyncer/v2/local_registry.go

```diff
@@ -0,0 +1,138 @@
+package v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	capabilities_registry_v2 "github.com/smartcontractkit/chainlink-evm/gethwrappers/workflow/generated/capabilities_registry_wrapper_v2"
+)
+
+type DonID uint32
+
+type DON struct {
+	capabilities.DON
+	CapabilityConfigurations map[string]CapabilityConfiguration
+}
+
+type CapabilityConfiguration struct {
+	Config []byte
+}
+
+type Capability struct {
+	ID             string
+	CapabilityType capabilities.CapabilityType
+}
+
+// LocalRegistry is a wrapper around mapping of IDs to the corresponding objects.
+type LocalRegistry struct {
+	lggr              logger.Logger
+	getPeerID         func() (types.PeerID, error)
+	IDsToDONs         map[DonID]DON
+	IDsToNodes        map[types.PeerID]capabilities_registry_v2.INodeInfoProviderNodeInfo
```

Comment:
If this is the only reason we've created a new type here, let's do the conversion that's being done in NodeByPeerID up front so that we localize the contract specific data types inside the new registrysyncer implementation